tests/integration/pilot/analysis/analysis_test.go

```diff
@@ -106,41 +105,77 @@ func TestWorkloadEntryUpdatesStatus(t *testing.T) {
 				Revision: "",
 				Labels:   nil,
 			})
-			// Make a workloadentry
-			ctx.Config().ApplyYAMLOrFail(t, ns.Name(), `
+
+			// create WorkloadEntry
+		ctx.Config().ApplyYAMLOrFail(t, ns.Name(), `
 apiVersion: networking.istio.io/v1alpha3
 kind: WorkloadEntry
 metadata:
   name: vm-1
+spec:
+  address: 127.0.0.1
 `)
-			// Test adding one condition
-			stat := &v1alpha1.IstioStatus{
-				Conditions: []*v1alpha1.IstioCondition{
+
+			retry.UntilSuccessOrFail(t, func() error {
+				// we should expect an empty array not nil
+				return expectWorkloadEntryStatus(t, ctx, ns, nil)
+			})
+
+			// add one health condition and one other condition
+			addedConds := []*v1alpha1.IstioCondition{
+				{
+					Type:   "Health",
+					Reason: "DontTellAnyoneButImNotARealReason",
+					Status: "True",
+				},
+				{
+					Type:   "SomeRandomType",
+					Reason: "ImNotHealthSoDontTouchMe",
+					Status: "True",
+				},
+
+			}
+
+			// Get WorkloadEntry to append to
+			we, err := ctx.Clusters().Default().Istio().NetworkingV1alpha3().WorkloadEntries(ns.Name()).Get(context.TODO(), "vm-1", metav1.GetOptions{})
+
+			if err != nil {
+				t.Error(err)
+			}
+
+			if we.Status.Conditions == nil {
+				we.Status.Conditions = []*v1alpha1.IstioCondition{}
+			}
+			// append to conditions
+			we.Status.Conditions = append(we.Status.Conditions, addedConds...)
+			// update the status
+			we, err = ctx.Clusters().Default().Istio().NetworkingV1alpha3().WorkloadEntries(ns.Name()).UpdateStatus(context.TODO(), we, metav1.UpdateOptions{})
+			if err != nil {
+				t.Error(err)
+			}
+			// we should have all the conditions present
+			retry.UntilSuccessOrFail(t, func() error {
+				// should update
+				return expectWorkloadEntryStatus(t, ctx, ns, []*v1alpha1.IstioCondition{
 					{
-						Type: "Health",
+						Type:   "Health",
 						Reason: "DontTellAnyoneButImNotARealReason",
 						Status: "True",
 					},
 					{
-						Type: "SomeRandomType",
+						Type:   "SomeRandomType",
 						Reason: "ImNotHealthSoDontTouchMe",
 						Status: "True",
-					},
-				},
-			}
-			ctx.Clusters().Default().Istio().NetworkingV1alpha3().WorkloadEntries(ns.Name()).UpdateStatus(context.TODO(), &clientnetworkingv1alpha3.WorkloadEntry{
-				Status:  *stat,
+					}})
+			})
 
-			}, metav1.UpdateOptions{})
-			retry.UntilSuccessOrFail(t, func() error {
-				// should update
-				return expectWorkloadEntryStatus(t, ctx, ns, stat.Conditions)
-			}, retry.Timeout(time.Minute*5))
+			// get the workload entry to replace the health condition field
+			we, err = ctx.Clusters().Default().Istio().NetworkingV1alpha3().WorkloadEntries(ns.Name()).Get(context.TODO(), "vm-1", metav1.GetOptions{})
 
 			// replacing the condition
-			for i, cond := range stat.Conditions {
+			for i, cond := range we.Status.Conditions {
```

Comment:
This logic should be abstracted into a public function, to ensure this pattern is used for updating conditions in the feature.

Reply:
I think adding that function once healthchecking is being implemented and replacing this logic with that function might be the better move. I might be wrong.